File: skip-the-dishes-query-resource/src/main/java/br/com/skip/the/dishes/query/resource/product/ProductApi.java
package br.com.skip.the.dishes.query.resource.product;

import br.com.skip.the.dishes.query.repository.product.ProductEntity;
import br.com.skip.the.dishes.query.resource.Paths;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequestMapping(path = Paths.PRODUCT_PATH)
public interface ProductApi {

    @GetMapping
    ResponseEntity<List<ProductEntity>> findAll(@RequestParam(name = "page", required = false) Integer page,
                                                @RequestParam(name = "size", required = false) Integer size,
                                                @RequestParam(name = "fields", required = false) String fields);

    @ResponseStatus(HttpStatus.OK)
    @GetMapping(path = "/{id}")
    ResponseEntity<ProductEntity> findById(@PathVariable("id") String id);

}

File: skip-the-dishes-query-resource/src/main/java/br/com/skip/the/dishes/query/resource/product/ProductApi.kt
package br.com.skip.the.dishes.query.resource.product

import br.com.skip.the.dishes.query.repository.product.ProductEntity
import br.com.skip.the.dishes.query.resource.Paths
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RequestMapping(path = [(Paths.PRODUCT_PATH)])
interface ProductApi {

    companion object {
        const val DEFAULT_PAGE = 0

        const val DEFAULT_SIZE = 20
    }

    @GetMapping
    fun findAll(@RequestParam(name = "page", defaultValue = DEFAULT_PAGE.toString()) page: Int,
                @RequestParam(name = "size", defaultValue = DEFAULT_SIZE.toString()) size: Int,
                @RequestParam(name = "fields", defaultValue = "") fields: String): ResponseEntity<List<ProductEntity>>

    @ResponseStatus(HttpStatus.OK)
    @GetMapping(path = ["/{id}"])
    fun findById(@PathVariable("id") id: String): ResponseEntity<ProductEntity>

}
